src/arduino.cc/arduino-builder/main.go

```diff
@@ -249,8 +248,7 @@ func main() {
 		_, err := os.Stat(buildPath)
 		if err != nil {
 			fmt.Fprintln(os.Stderr, err)
-			defer os.Exit(1)
-			return
+			os.Exit(1)
```

Comment:
I guess you could even use a more simple approach where the effective main function can just return an exit code, just like we know from C:

```
func main() {
    os.Exit(realMain())
}
```

And then the `realMain()` function just does what it needs and returns an exit code?